sympy/matrices/reductions.py

```diff
@@ -314,6 +318,47 @@ def _rref(M, iszerofunc=_iszero, simplify=False, pivots=True,
     >>> rref_pivots
     (0, 1)
 
+    If pivots is ``False`` just the row-reduced matrix is returned.
+
+    >>> m.rref(pivots=False)
+    Matrix([
+    [1, 0],
+    [0, 1]])
+
+    If different iszerofunc is used such as ``lambda x: abs(x)<6e-15``.
+
+    >>> m = Matrix([[0.9, -0.1, -0.2, 0],[-0.8, 0.9, -0.4, 0],[-0.1, -0.8, 0.6, 0]])
+    >>> m.rref(iszerofunc=lambda x: abs(x)<6e-15)
+    (Matrix([
+    [1, 0, -0.301369863013699, 0],
+    [0, 1, -0.712328767123288, 0],
+    [0, 0, 0, 0]]), (0, 1))
+
+    If dotprodsimp is set to ``True``.
+
+    >>> M = Matrix(3, 3, [x+i for i in range (9)])
+    >>> M.rref(dotprodsimp=True)

```

Comment:
I think we need an example that demonstrates the effect of dotprodsimp e.g. by comparing the result of  rref with rref(dotprodsimp=True). Also the sentence should explain what it does.